src/metabase/driver/ddl/interface.clj

```diff
@@ -0,0 +1,58 @@
+(ns metabase.driver.ddl.interface
+  (:require [clojure.string :as str]
+            [metabase.driver :as driver]
+            [metabase.util.i18n :refer [tru]]))
+
+(defn schema-name
+  "Returns a schema name for persisting models. Needs the database to use the db id and the site-uuid to ensure that
+  multiple connections from multiple metabae remain distinct. The UUID will have the first character of each section taken.
+
+  (schema-name {:id 234} \"143dd8ce-e116-4c7f-8d6d-32e99eaefbbc\") ->  \"metabase_cache_1e483_1\""
+  [{:keys [id] :as _database} uuid-string]
+  (let [instance-string (apply str (map first (str/split uuid-string #"-")))]
+    (format "metabase_cache_%s_%s" instance-string id)))
+
+(defmulti format-name
+  "Transform a lowercase string Table or Field name in a way appropriate for this dataset (e.g., `h2` would want to
+  upcase these names; `mongo` would want to use `\"_id\"` in place of `\"id\"`. This method should return a string.
+  Defaults to an identity implementation."
+  {:arglists '([driver table-or-field-name])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
+(defmethod format-name :default [_ table-or-field-name] table-or-field-name)
+
+(defmulti field-base-type->sql-type
+  "A suitable db type for a base-type per database."
+  {:arglists '([driver base-type])}
+  (fn [driver base-type] [driver base-type])
```

Comment:
Does this need to (maybe) initialize `driver`? (there's always a chance we can be calling this for say `:sqlite` and it will fail because the `:sqlite` driver (and thus the impl for this method) have not been loaded

Also I would vote on doing `(keyword base-type)` because `defmethod` dispatch adds an entry for every value returned by the dispatch function. So it's important that the set of values returned by the dispatch function is bounded. Since keywords get permanently interned the set of keywords is implicitly bounded at some point

Reply:
Good points on both. My thinking was that if you start persisting the driver is obviously already initialized. But you're right, on restart a persist could be the first operation on the db. Good points on both.